### PR TITLE
INFRA: Upgrade All Provision Code to Utilize Latest Azure Libraries 

### LINKED ISSUE #25: INFRA: Upgrade All Provision Code to Utilize Latest Azure Libraries
Assigned to @Pwongchaiya

### PR DESCRIPTION
**Pull request by Pwongchaiya**
Closes #25

### CODE REVIEW

**Comment by Pwongchaiya** [comment]
Screenshot 2023-05-05 at 22-15-56 Microsoft Azure1
Screenshot (125)

File: InternTrack.Core.Api.Infrastructure.Provision/Brokers/Clouds/CloudBroker.cs [hunk 1]
```diff
@@ -3,45 +3,31 @@
...
-using Microsoft.Azure.Management.ResourceManager.Fluent;
-using Microsoft.Azure.Management.ResourceManager.Fluent.Authentication;
-using Microsoft.Azure.Management.ResourceManager.Fluent.Core;
+using Azure.Identity;
+using Azure.ResourceManager;
+
+/* 
+Environment Variables naming convention:
+AZURE_CLIENT_ID
+AZURE_CLIENT_SECRET
```
Comment (hassanhabib) [inline InternTrack.Core.Api.Infrastructure.Provision/Brokers/Clouds/CloudBroker.cs:12]: No comments

File: InternTrack.Core.Api.Infrastructure.Provision/Brokers/Clouds/CloudBroker.Plans.cs [hunk 2]
```diff
@@ -3,26 +3,45 @@
...
-                .Define(planName)
-                .WithRegion(Region.USWest3)
-                .WithExistingResourceGroup(resourceGroup)
-                .WithPricingTier(PricingTier.StandardS1)
-                .WithOperatingSystem(OperatingSystem.Windows)
-                .CreateAsync();
+            AppServicePlanCollection appServicePlans = resourceGroup
+                .GetAppServicePlans();
+
+            AppServicePlanData appServicePlanData =
```
Comment (hassanhabib) [inline InternTrack.Core.Api.Infrastructure.Provision/Brokers/Clouds/CloudBroker.Plans.cs:24]: If the right side type is explicit then use `var` on the left.
For instance:
```
var student = new Student();
```

Don't do:
```
Student student = new Student();
```

Fix all others the same way

**Comment by hassanhabib** [comment]
@Pwongchaiya where's AppInsights? :)

**Pull request merged**